pac4j-oauth/src/main/java/org/pac4j/oauth/client/FacebookClient.java

```diff
@@ -201,6 +212,50 @@ protected boolean hasBeenCancelled(final WebContext context) {
             return false;
         }
     }
+
+    public void setUseAppSecretProof(boolean useSecret) {
+        this.useAppsecretProof = useSecret;
+    }
+
+    public boolean getUseAppSecretProof() {
+        return this.useAppsecretProof;
+    }
+
+    /**
+     * The code in this method is based on this blog post: https://www.sammyk.me/the-single-most-important-way-to-make-your-facebook-app-more-secure
+     * and this answer: https://stackoverflow.com/questions/7124735/hmac-sha256-algorithm-for-signature-calculation
+     *
+     * @param url the URL to which we're adding the proof
+     * @param token the application token we pass back and forth
+     * @return URL with the appsecret_proof parameter added
+     */
+    protected String computeAppSecretProof(String url, Token token) {
+        try {
+            Mac sha256_HMAC = Mac.getInstance("HmacSHA256");
+            SecretKeySpec secret_key = new SecretKeySpec(this.secret.getBytes("UTF-8"), "HmacSHA256");
+            sha256_HMAC.init(secret_key);
+            String proof = org.apache.commons.codec.binary.Hex.encodeHexString(sha256_HMAC.doFinal(token.getToken().getBytes("UTF-8")));
+            url = CommonHelper.addParameter(url, APPSECRET_PARAMETER, proof);
```

Comment:
This assignment is not really necessary and we could just return the String returned by CommonHelper.addParameter(...). This is actually a leftover from debugging.